src/mrpro/operators/functionals/L1NormViewAsReal.py

```diff
@@ -0,0 +1,84 @@
+"""L1 Norm."""
+
+import torch
+
+from mrpro.operators.Functional import ProximableFunctional
+
+
+class L1NormViewAsReal(ProximableFunctional):
```

Comment:
moreau identity. uses the prox. a explicit implementation was not much faster and replicated the is-complex logic from the prox, so we just use the base classes implemntation

Reply:
ah yeah, I had missed that indeed.

one general thought: isn't this something that  in general we should avoid to do? What I mean is that the Moreau identity is some sort of result, similar to 1=sin^2(x) + cos^2(x).

So if for a functional one derives a prox operator that is wrong but for some reason still passes all the tests (could this happen?), one would define the prox_conv_conj by using the wrong prox. However, the Moreau-identity would be fulfilled by construction. Like defining, the cosine by cos^2(x) = 1- sin^2(x) instead of cos(x) = adjacent/hypotenuse and having 1=sin^2(x) + cos^2(x) as a result.

Reply:
but most of the codes to actually calculate the sin/cos use these properties to calculate it :)

it is equally likely to have a wrong prox_convex_conj that passes moreau identity as having a wrong prox. imho it is not a bug but a feature that prox_convex_conj will automatically be correct if prox is correct.

it only makes sense to manually implement it if there are some simplifications that can be done in the code. 
but here, the dtype checks would be most of the code of the prox_convex_conj.

Reply:
ok, but sin/cos is much more trivial than the case here. I would have thought that it is very unlikely to be able to implement a (probably wrong) prox_conv_conj that compensates for the wrongly implemented prox such that the Moreau identity passes. Thus, I thought that, whenever possible, it could be better to have both implementations and maybe for the case where one of prox or prox_conv_conv is much more computationally intensive than the other, to rely on the Moreau-identity to compute the desired operator.

And sure, I also agree the Moreau identity is a "feature", I was just wondering if coding-wise it might potentially lead to wrong implementations.